app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -31,12 +35,65 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
             return value;
         }
 
-        function handleFiltersUpdate(filters) {
+        /**
+         * Looks for keys that are NOT present in currentFilters but that are in the savedSearch filters (which means they are removed)
+         * and removes them from the saved search filters array.
+         * Does not handle removal where the key exists but the values are an array and some are missing. Need to fix that.
+         **/
+        function cleanDeprecatedValuesFromSavedSearch(currentFilters, savedSearch) {
+            //find filters in currentFilters that are NOT in savedSearch.filters
```

Comment:
strong case for a decent diffing library for sure. :/

Reply:
also I'm changing this name before it lands , thinking a bit more about the whole thing